test/system/projects/samples_test.rb

```diff
@@ -347,21 +347,32 @@ class SamplesTest < ApplicationSystemTestCase
       assert_no_text @sample3.name
     end
 
-    test 'can sort the list of samples' do
+    test 'can sort samples by column' do
       visit namespace_project_samples_url(namespace_id: @namespace.path, project_id: @project.path)
 
       assert_selector 'table#samples-table tbody tr', count: 3
       within first('tbody tr td:nth-child(2)') do
         assert_text @sample1.name
       end
 
-      click_on I18n.t(:'projects.samples.index.sorting.updated_at_desc')
-      click_on I18n.t(:'projects.samples.index.sorting.name_desc')
+      click_on I18n.t('projects.samples.table.sample')
 
-      assert_text I18n.t(:'projects.samples.index.sorting.name_desc')
+      assert_selector 'table thead th:nth-child(2) svg.icon-arrow_up'
       assert_selector 'table#samples-table tbody tr', count: 3
-      within first('tbody tr td:nth-child(2)') do
-        assert_text @sample3.name
+      within first('tbody') do
+        assert_selector 'tr:first-child td:nth-child(2)', text: @sample1.name
+        assert_selector 'tr:nth-child(2) td:nth-child(2)', text: @sample2.name
+        assert_selector 'tr:last-child td:nth-child(2)', text: @sample3.name
+      end
+
+      click_on I18n.t('projects.samples.table.sample')
+
+      assert_selector 'table thead th:nth-child(2) svg.icon-arrow_down'
+      assert_selector 'table#samples-table tbody tr', count: 3
+      within first('tbody') do
+        assert_selector 'tr:first-child td:nth-child(2)', text: @sample3.name
+        assert_selector 'tr:nth-child(2) td:nth-child(2)', text: @sample2.name
+        assert_selector 'tr:last-child td:nth-child(2)', text: @sample1.name
       end
     end
```

Comment:
Can you also add tests for sorting by the other default columns of the table which have sorting enabled?

Reply:
Updated in 4841c4c